src/mapgen_functions.cpp

```diff
@@ -977,382 +980,319 @@ void mapgen_fungal_flowers(map *m, oter_id, mapgendata dat, int, float)
     m->add_spawn(mon_fungaloid_seeder, 1, 12, 12);
 }
 
-void mapgen_road_straight(map *m, oter_id terrain_type, mapgendata dat, int turn, float density)
-{
-    bool sidewalks = false;
-    for (int i = 0; i < 8; i++) {
-        if (otermap[dat.t_nesw[i]].has_flag(has_sidewalk)) {
-            sidewalks = true;
+int terrain_type_with_suffix_to_nesw_array(oter_id terrain_type, bool array[4]) {
+    // extract the suffix from the terrain type name
+    std::string suffix = std::string(terrain_type).substr( std::string(terrain_type).find_last_of("_") + 1 );
+    // non-"_end" end tiles have _north _east _south _west, all of which contain "t"
+    if (suffix.find("t") != std::string::npos) {
+        suffix = suffix.substr(0,1);
+        if      (suffix=="n") suffix = "s"; // and they are all backwards :(
+        else if (suffix=="e") suffix = "w";
+        else if (suffix=="s") suffix = "n";
+        else if (suffix=="w") suffix = "e";
```

Comment:
Same here. A simple function to invert the letter invoked 4 times wouldn't get extended to 3 lines with astyle and would look more like the rest of the code.